src/earn/EarnEnterAmount.tsx

```diff
@@ -0,0 +1,626 @@
+import EarnDepositBottomSheet from 'src/earn/EarnDepositBottomSheet'
+import { fetchAavePoolInfo } from 'src/earn/poolInfo'
+import { usePrepareSupplyTransactions } from 'src/earn/prepareTransactions'
+import InfoIcon from 'src/icons/InfoIcon'
+import { LocalCurrencySymbol } from 'src/localCurrency/consts'
+import { getLocalCurrencySymbol } from 'src/localCurrency/selectors'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import { useSelector } from 'src/redux/hooks'
+import { AmountInput, ProceedArgs } from 'src/send/EnterAmount'
+import { AmountEnteredIn } from 'src/send/types'
+import Colors from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import variables from 'src/styles/variables'
+import { useLocalToTokenAmount, useTokenInfo, useTokenToLocalAmount } from 'src/tokens/hooks'
+import { feeCurrenciesSelector } from 'src/tokens/selectors'
+import { TokenBalance } from 'src/tokens/slice'
+import Logger from 'src/utils/Logger'
+import networkConfig, { networkIdToNetwork } from 'src/web3/networkConfig'
+import { walletAddressSelector } from 'src/web3/selectors'
+import { Address, isAddress } from 'viem'
+
+type Props = NativeStackScreenProps<StackParamList, Screens.EarnEnterAmount>
+
+const TAG = 'EarnEnterAmount'
+
+const TOKEN_SELECTOR_BORDER_RADIUS = 100
+const MAX_BORDER_RADIUS = 96
+const FETCH_UPDATED_TRANSACTIONS_DEBOUNCE_TIME = 250
+
+type ProceedComponentProps = Omit<ProceedArgs, 'tokenAmount'> & {
+  onPressProceed(args: ProceedArgs): void
+  onPressInfo(): void
+  disabled: boolean
+  tokenAmount: BigNumber | null
+}
+
+function EarnEnterAmount({ route }: Props) {
+  const { t } = useTranslation()
+
+  const { tokenId } = route.params
+  const token = useTokenInfo(tokenId)
+
+  if (!token) {
+    throw new Error(`Token info not found for token ID ${tokenId}`)
+  }
+
+  const infoBottomSheetRef = useRef<BottomSheetRefType>(null)
+  const addCryptoBottomSheetRef = useRef<BottomSheetRefType>(null)
+  const reviewBottomSheetRef = useRef<BottomSheetRefType>(null)
+  const tokenAmountInputRef = useRef<RNTextInput>(null)
+  const localAmountInputRef = useRef<RNTextInput>(null)
+
+  const [tokenAmountInput, setTokenAmountInput] = useState<string>('')
+  const [localAmountInput, setLocalAmountInput] = useState<string>('')
+  const [enteredIn, setEnteredIn] = useState<AmountEnteredIn>('token')
+  // this should never be null, just adding a default to make TS happy
+  const localCurrencySymbol = useSelector(getLocalCurrencySymbol) ?? LocalCurrencySymbol.USD
+
+  const {
+    prepareTransactionsResult,
+    refreshPreparedTransactions,
+    clearPreparedTransactions,
+    prepareTransactionError,
+  } = usePrepareSupplyTransactions()
+
+  const walletAddress = useSelector(walletAddressSelector) as Address
```

Comment:
can we avoid the as Address, I think we've been doing a `isAddress` check where required